### PR TITLE
Rule by methods

### PR DESCRIPTION
**Pull request by g4s8**
#220 - refactored routing rule by methods

### CODE REVIEW

File: src/main/java/com/artipie/http/rt/RlByMethods.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Route by HTTP methods rule.
+ * @since 0.16
+ */
+public final class RlByMethods implements RtRule {
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/rt/RlByMethods.java:38]: @g4s8 guess there is a typo and class should be named `RtByMethod`, not `RlByMethod`

File: src/main/java/com/artipie/http/rt/RlByMethods.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+     */
+    public RlByMethods(final Set<RqMethod> methods) {
+        this.methods = Collections.unmodifiableSet(methods);
+    }
+
+    @Override
+    public boolean apply(final String line,
+        final Iterable<Map.Entry<String, String>> headers) {
+        return this.methods.contains(new RequestLineFrom(line).method());
+    }
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/rt/RlByMethods.java:115]: @g4s8 I'd suggest to add a test for this method

**Comment by g4s8** [comment]
@olegmoz thanks, fixed in https://github.com/artipie/http/pull/221/commits/0886ad5db6108e0c00ecc9aef795a7486ab5e1c4

**Comment by 0crat** [comment]
Job `gh:artipie/http#221` is not assigned, can't get performer

**Pull request merged**